### DISCUSSION TITLE
posts/2023-04-24-oh-my-god-too-many-open-files/

### DISCUSSION THREAD

**Discussion by giscus** [Hanaasagi/blog#6] [Announcements]
# posts/2023-04-24-oh-my-god-too-many-open-files/

Hanaasagi's personal blog

https://blog.dreamfever.me/posts/2023-04-24-oh-my-god-too-many-open-files/

**Comment by chasing-freedom** [comment]
你好，博主。我用V2raya+codeium.nvim 也遇到了这个问题，请问你后面是怎么解决的？用旁路由等外部代理来代替本地透明代理么？

**Reply by chasing-freedom** [reply]
又看了看博文，应该是对这个网址添加白名单，不代理？

**Reply by chasing-freedom** [reply]
查了下仓库，有国人老哥也注意到了这个问题并提交了一个 [pr](https://github.com/Exafunction/codeium.nvim/pull/123)，不过被关闭了。
维护者说用户添加 curl 排除项来取消代理。对于透明代理来说，估计还是要代理软件那边添加直连白名单。

**Reply by Hanaasagi** [reply]
```diff
diff --git a/autoload/codeium/server.vim b/autoload/codeium/server.vim
index ff9c1c4..8571845 100644
--- a/autoload/codeium/server.vim
+++ b/autoload/codeium/server.vim
@@ -71,6 +71,7 @@ function! codeium#server#Request(type, data, ...) abort
   let args = [
               \ 'curl', uri,
               \ '--header', 'Content-Type: application/json',
+              \ '--max-time', '1',
               \ '-d@-'
               \ ]
   let result = {'out': [], 'err': []}
```

@chasing-freedom 可以先尝试加一个超时。 curl 会在超时后主动关闭。
